src/extension.ts

```diff
@@ -134,22 +33,42 @@ export function activate(context: ExtensionContext) {
 
   // Create the language client and start the client.
 
-  client = new LanguageClient(
-    "smithyLsp",
-    "Smithy LSP",
-    createServer,
-    clientOptions
-  );
-  const smithyContentProvider = createSmithyContentProvider(client);
-  context.subscriptions.push(
-    workspace.registerTextDocumentContentProvider(
-      "smithyjar",
-      smithyContentProvider
-    )
-  );
+  const version = vscode.workspace
+    .getConfiguration("smithyLsp")
+    .get("version", "`");
+
+  const lspCoordinates = vscode.workspace
+    .getConfiguration("smithyLsp")
+    .get("lspCoordinates", "`");
+
+  return getCoursierExecutable(context.globalStoragePath).then(
+    (csBinaryPath) => {
+      console.info(`Resolved coursier's binary at ${csBinaryPath}`);
+
+      const startServer = {
+        command: csBinaryPath,
+        args: ["launch", `${lspCoordinates}:${version}`, "--", "0"],
```

Comment:
add `--mode force` here

Reply:
I did it but I don't see why it would be necessary

It's even possible that with that flag, a fully bootstrapped project (meaning you've opened it before, the LSP was started before, everything is in cache) fails to start if you're offline

Reply:
I'd suggest `-ttl 0`, that way only the "latest.release" thing is forced to be resolved, and you don't necessarily prune the whole cache for the jars

Reply:
personally I launch my LSP with `--ttl 1h`, and for local development times I would possibly change that to 0.

Reply:
@daddykotex you're right, before the `--mode force` was run on a `resolve` command, which didn't actually induce the download. Let's go with Jakub's solution

Reply:
done